ibis/backends/base/sql/__init__.py

```diff
@@ -132,6 +132,32 @@ def raw_sql(self, query: str) -> Any:
     def _safe_raw_sql(self, *args, **kwargs):
         yield self.raw_sql(*args, **kwargs)
 
+    @staticmethod
+    def _check_pyarrow_install():
+        try:
+            import pyarrow as pa  # noqa: ignore
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Exporting to Arrow formats requires `pyarrow` but it is not installed"  # noqa: ignore
+            )
+
+    def to_pyarrow_batches(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        chunk_size: int = 1_000_000,
+    ):
+        raise NotImplementedError
+
+    def to_pyarrow(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+    ):
+        raise NotImplementedError
```

Comment:
A default implementation could create the table from the result of `to_pyarrow_batches()`

Reply:
Ahh, yeah, that's what I'm doing in the `AlchemyBackend` but I can move that bit one step up the class hieararchy